nixos/lib/systemd-unit-options.nix

```diff
@@ -338,6 +349,11 @@ in rec {
         configuration switch if its definition has changed.  If
         enabled, the value of <option>restartIfChanged</option> is
         ignored.
+
+        This option should not be used anymore in favor of
+        <option>reloadTriggers</option> which allows more granular
+        control of when a service is reloaded and when a service
+        is restarted.
```

Comment:
We seem to be only having 2 instances of this in nixpkgs. I'd rather see this removed, and migration guides be shown in the release notes.

Reply:
I'd go for fixing the instances and hiding this option for the 22.05 release instead of removing it.
This way we ge around the annoying case where people who are using `reloadIfChanged` have to migrate it while migrating to 22.05 because the new option doesn't exist in 21.11 and the old one doesn't exist on 22.05.
When hiding the option and removing it for 22.11, people have the full 22.05 release to migrate.